internal/lock.go

```diff
@@ -49,13 +55,13 @@ func NewLock(path string, newOk bool) (*Lock, error) {
 	return &Lock{path: path, conf: conf}, nil
 }
 
-func (l *Lock) AddResource(paths []string, algo string, tags []string, filename string) error {
+func (l *Lock) AddResource(paths []string, algo string, tags []string, filename string, dynamic bool) error {
```

Comment:
I don't see why you need to add this parameter. I think you can simply remove it. Here and everywhere it's "used".